vault/resource_gcp_secret_impersonated_account_test.go

```diff
@@ -0,0 +1,191 @@
+package vault
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"github.com/hashicorp/terraform-provider-vault/testutil"
+	"golang.org/x/oauth2/google"
+)
+
+// This test requires that you pass credentials for a user or service account having the IAM rights
+// listed at https://www.vaultproject.io/docs/secrets/gcp/index.html for the project you are testing
+// on. The credentials must also allow setting IAM permissions on the project being tested.
+func TestGCPSecretImpersonatedAccount(t *testing.T) {
+	backend := "gcp"
+	impersonatedAccount := acctest.RandomWithPrefix("tf-test")
+	credentials, project := testutil.GetTestGCPCreds(t)
+
+	// We will use the provided key as the impersonated account
+	conf, err := google.JWTConfigFromJSON([]byte(credentials), "https://www.googleapis.com/auth/cloud-platform")
+	if err != nil {
+		t.Fatalf("error decoding GCP Credentials: %v", err)
+	}
+	serviceAccountEmail := conf.Email
+
+	noBindings := testGCPSecretImpersonatedAccount_accessToken(backend, impersonatedAccount, credentials, serviceAccountEmail)
+
+	resourceName := "vault_gcp_secret_impersonated_account.test"
+	resource.Test(t, resource.TestCase{
+		Providers:    testProviders,
+		PreCheck:     func() { testutil.TestAccPreCheck(t) },
+		CheckDestroy: testGCPSecretImpersonatedAccountDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: noBindings,
+				Check: resource.ComposeTestCheckFunc(
+					testGCPSecretImpersonatedAccount_attrs(backend, impersonatedAccount),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldBackend, backend),
+					resource.TestCheckResourceAttr(resourceName, "impersonated_account", impersonatedAccount),
+					resource.TestCheckResourceAttr(resourceName, "service_account_email", serviceAccountEmail),
+					resource.TestCheckResourceAttr(resourceName, "service_account_project", project),
+					resource.TestCheckResourceAttr(resourceName, "token_scopes.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "token_scopes.0", "https://www.googleapis.com/auth/cloud-platform"),
+				),
+			},
+			{
+				ResourceName:            "vault_gcp_secret_impersonated_account.test",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{},
+			},
+		},
+	})
+}
+
+func testGCPSecretImpersonatedAccount_attrs(backend, impersonatedAccount string) resource.TestCheckFunc {
```

Comment:
as described in a comment above, we can remove this function as it is no longer needed in our testing loop